### PR TITLE
added inheritance

### PR DESCRIPTION
**Pull request by notsure**

### CODE REVIEW

**Comment by 3rfan** [comment]
Ack

File: src/fox.js [hunk 1]
```diff
@@ -29,19 +32,14 @@
...
-            baseService = this.services[baseServiceClass[0]];
-            if (baseService) {
-                Service = extend(Service, baseService);
-            } else {
-                console.error('Could not extend your base service: "' + baseServiceClass + '".');
-            }
-        }
-        this.services[name] = new Service;
+    App.prototype.registerService = function(dependencies, name, serviceClass) {
+        var service = getExtendedClass(AbstractService,
```
Comment (chirimoya) [inline src/fox.js]: classes are written in UpperCamelCase?

File: src/fox.js [hunk 2]
```diff
@@ -81,12 +79,26 @@
...
         this.events = new EventEmitter();
-    }
+    };
 
     var ownProp = function(obj, key) {
         return Object.prototype.hasOwnProperty.call(obj, key);
     };
 
+    var getExtendedClass = function(AbstractBaseClass, trunk, deps, className, classObject) {
+        var extClass = AbstractBaseClass;
```
Comment (chirimoya) [inline src/fox.js]: classes are written in UpperCamelCase?

File: src/fox.js [hunk 3]
```diff
@@ -81,12 +79,26 @@
...
     var AbstractService = function() {
         this.events = new EventEmitter();
-    }
+    };
 
     var ownProp = function(obj, key) {
         return Object.prototype.hasOwnProperty.call(obj, key);
     };
 
+    var getExtendedClass = function(AbstractBaseClass, trunk, deps, className, classObject) {
```
Comment (chirimoya) [inline src/fox.js]: I wouldn't shorten dependencies. Any uglify will do it anyway.

**Comment by chirimoya** [comment]
else ACK

**Pull request merged**